src/mono/CMakeLists.txt

```
@@ -204,6 +204,7 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set(HOST_LINUX 1)
   add_definitions(-D_GNU_SOURCE -D_REENTRANT)
   add_definitions(-D_THREAD_SAFE)
+  add_link_options(-Wl,-z,relro,-z,now)
```

can we make this consistent with Android:

```suggestion
  add_compile_options(-Wl,-z,now)
  add_compile_options(-Wl,-z,relro)
```

... what about  `noexecstack`?

---

ah I just saw https://github.com/dotnet/runtime/pull/59987#issuecomment-934385040 and you tried that already. it's interesting why this doesn't work on Linux, can you please still split it into different lines and leave a comment about why add_link_options is used?

---

The "add_compile_options" didn't actually work for me - it is not being applied at the final *link* step, which is actually the only step where it really matters.

As to noexecstack, the library is already marked as noexecstack to me.   That's to be expected as the compilers already mark objects files they generate accordingly.  The only reason for a manual override is when you link in a full assembler file (.s), which I don't believe happens in the Mono build.

---

Sorry, didn't see the latest comment.   I'm actually a bit surprised that add_compile_options works for Android, I would have expected the same problem (I can't really test that here).   If you prefer two lines, I can certainly change that, I just used what was in eng/native/configurecompiler.cmake  ...

---

Added comment and split back into two lines.   Let me know if there's anything else you'd like me to do.   Thanks!

---

Thanks, looks good now. We can probably change Android to use `add_link_options` too, I suppose `add_compile_options` works because we use clang there.